shell/src/main/java/alluxio/cli/fsadmin/pathconf/ShowCommand.java

```diff
@@ -0,0 +1,143 @@
+ * 2. show all configurations applicable to the path.
+ * <p>
+ * For example, suppose there are two path level configurations:
+ * /a: property1=value1
+ * /a/b: property2=value2
+ * Then in mode 1, only property2=value2 will be shown for /a/b,
+ * but in mode 2, since /a is the prefix of /a/b, both properties will be shown for /a/b.
+ */
+public final class ShowCommand extends AbstractFsAdminCommand {
+  public static final String RESOLVE_OPTION_NAME = "resolve";
+
+  private static final Option RESOLVE_OPTION =
+      Option.builder()
+          .longOpt(RESOLVE_OPTION_NAME)
+          .required(false)
+          .hasArg(false)
+          .desc("Whether to show all path level configurations applicable to the path")
+          .build();
+
+  /**
+   * @param context     fsadmin command context
+   * @param alluxioConf Alluxio configuration
+   */
+  public ShowCommand(Context context, AlluxioConfiguration alluxioConf) {
+    super(context);
+  }
+
+  @Override
+  public String getCommandName() {
+    return "show";
+  }
+
+  @Override
+  public Options getOptions() {
+    return new Options()
+        .addOption(RESOLVE_OPTION);
+  }
+
+  @Override
+  public void validateArgs(CommandLine cl) throws InvalidArgumentException {
+    CommandUtils.checkNumOfArgsNoLessThan(this, cl, 1);
+  }
+
+  private String format(String key, String value) {
+    return key + " = " + value;
+  }
+
+  @Override
+  public int run(CommandLine cl) throws IOException {
+    String targetPath = cl.getArgs()[1];
+    Configuration configuration = mMetaConfigClient.getConfiguration();
+
+    if (cl.hasOption(RESOLVE_OPTION_NAME)) {
+      Set<PropertyKey> pathPropertyKeys = new HashSet<>();
+      Map<String, AlluxioConfiguration> pathConfMap = new HashMap<>();
+
+      configuration.getPathConf().forEach((path, conf) -> {
+        AlluxioProperties properties = new AlluxioProperties();
+        conf.forEach(property -> {
+          pathPropertyKeys.add(PropertyKey.fromString(property.getName()));
+          properties.set(PropertyKey.fromString(property.getName()),
+              property.getValue());
+        });
+        pathConfMap.put(path, new InstancedConfiguration(properties));
+      });
+      PathConfiguration pathConf = PathConfiguration.create(pathConfMap);
+
+      AlluxioURI targetUri = new AlluxioURI(targetPath);
+      List<PropertyKey> propertyKeys = new ArrayList<>(pathPropertyKeys);
+      propertyKeys.sort(Comparator.comparing(PropertyKey::getName));
+      propertyKeys.forEach(key -> {
+        pathConf.getConfiguration(targetUri, key).ifPresent(conf -> {
+          mPrintStream.println(format(key.getName(), conf.get(key)));
+        });
+      });
+    } else {
+      if (configuration.getPathConf().containsKey(targetPath)) {
+        List<Property> properties = configuration.getPathConf().get(targetPath);
+        properties.sort(Comparator.comparing(Property::getName));
+        properties.forEach(property -> {
```

Comment:
could move the sorting and printing outside the if/else

Reply:
consolidated into one `else if` block, I think it's more intuitive to put them together in one block